src/workspaces/ShareWorkspaceModal/Collaborator.test.ts

```diff
@@ -206,4 +206,83 @@ describe('a Collaborator component', () => {
     fireEvent.click(writterPermission);
     expect(setAcl).toHaveBeenCalledTimes(1);
   });
+
+  describe('only allows owners and project owners to share with additional permissions', () => {
+    const setAcl = jest.fn();
+    const aclMock = asMockedFn(setAcl);
+    const item: AccessEntry = {
+      email: 'user1@test.com',
+      pending: false,
+      canShare: true,
+      canCompute: true,
+      accessLevel: 'WRITER',
+    };
+    const acl = [item];
+
+    it('displays a tooltip when the user cannot share with additional permissions', () => {
+      // Arrange
+      render(
+        h(Collaborator, {
+          aclItem: item,
+          acl,
+          setAcl: aclMock,
+          originalAcl: acl,
+          workspace: { ...workspace, accessLevel: 'WRITER' },
+        })
+      );
+
+      // Act
+      const canShareCheckbox = getByRole(screen.getByText('Can share').parentElement!, 'checkbox');
+      fireEvent.mouseOver(canShareCheckbox);
+
+      // Assert
+      const tooltip = screen.getByRole('tooltip');
+      expect(tooltip).toHaveTextContent('Only Owners and Project Owners can share additional permissions');
+    });
+
+    test.each([
+      {
+        accessLevel: 'OWNER' as WorkspaceAccessLevel,
+        descriptor: 'allows',
+        shouldDisableCheckbox: false,
+      },
+      {
+        accessLevel: 'PROJECT_OWNER' as WorkspaceAccessLevel,
+        descriptor: 'allows',
+        shouldDisableCheckbox: false,
+      },
+      {
+        accessLevel: 'WRITER' as WorkspaceAccessLevel,
+        descriptor: 'does not allow',
+        shouldDisableCheckbox: true,
+      },
+    ])(
+      '$descriptor an $accessLevel to share with additional permissions',
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      ({ accessLevel, descriptor, shouldDisableCheckbox }) => {
```

Comment:
Nit: can we get rid of this ESLint control comment by removing `descriptor` from the destructure?

```suggestion
      ({ accessLevel, shouldDisableCheckbox }) => {
```

Reply:
Is it not necessary for the string interpolation?

Reply:
oh, looks like its not...

Reply:
It's still in the arguments table, so it's available for interpolation in the test name.

But the test function doesn't necessarily need to access it from its parameter.

It's a bit clearer to see if imagining this function without the inline destructure...

```js
it.each([
  ...
])(('$descriptor an $accessLevel to share with additional permissions', (args) => {
  const { accessLevel, shouldDisableCheckbox } = args;
  ...
```